core/src/main/java/org/testcontainers/lifecycle/TestLifecycleAware.java

```diff
@@ -0,0 +1,14 @@
+package org.testcontainers.lifecycle;
+
+import java.util.Optional;
+
+public interface TestLifecycleAware {
+
+    default void beforeTestBlock(TestDescription description) {
```

Comment:
Why `beforeTestBlock` and not `beforeTest`? What are the semantics of block?

Reply:
Just in case if some Test frameworks define `Test` as something more bigger and an atomic execution unit is some block of code. Arguable :)

Reply:
ok, I summon @rnorth as a tiebreaker 😄